packages/algob/src/lib/tx-log-writer.ts

```diff
@@ -0,0 +1,44 @@
+import fs from "fs";
+import path from "path";
+import YAML from "yaml";
+
+import { ARTIFACTS_DIR } from "../internal/core/project-structure";
+
+export interface txWriter {
+
+  timestamp: number
+  push: (scriptPath: string, msg: string, obj: any) => void
+  ensureDirectoryExistence: (filePath: string) => boolean
+
+}
+
+export class TxWriterImpl implements txWriter {
+  timestamp: number;
+
+  constructor () {
+    this.timestamp = +new Date();
+  }
+
+  ensureDirectoryExistence (filePath: string): boolean {
+    var dirname = path.dirname(filePath);
+    if (fs.existsSync(dirname)) {
+      return true;
+    }
+    this.ensureDirectoryExistence(dirname);
+    fs.mkdirSync(dirname);
+    return true;
+  }
+
+  push (scriptPath: string, msg: string, obj: any): void {
+    this.timestamp = +new Date();
+
+    const filePath = path.join(ARTIFACTS_DIR, scriptPath) + '.' +
+    (this.timestamp).toString() + '.log';
+
+    this.ensureDirectoryExistence(filePath);
+    fs.appendFileSync(
+      filePath,
+      msg + YAML.stringify(obj)
+    );
+  }
+}
```

Comment:
Please add a test for `TxWriterImpl`

To make it easier, you can parametrize the file writing with a function. For example `src/lib/compile.ts`. The `CompileOp` has a `writeFile` method which is overwritten in [tests](https://github.com/scale-it/algorand-builder/blob/master/packages/algob/test/builtin-tasks/compile.ts#L35). You can do it even simpler:

```
// in tests:
let txw = new  TxWriterImpl(...)
let writtenContent = []
txw.writeFile = (filename, content) => writtenContent.push({filename, content})

// then somewhere in tests
// check the writtenContent.
// add the beginning of a test you should reset writtenContent:
writtenContent = []
```

Reply:
ok